src/slurm_cloud_bursting/suspend.py

```diff
@@ -0,0 +1,174 @@
+
+from slurm_cloud_bursting.utils import (
+    CONFIG_FILE_PATH,
+    DEFAULT_MAX_INSTANCES_BATCH_SIZE,
+    DEFAULT_MAX_RETRY,
+    SCONTROL,
+    get_nodes_info,
+    grouper,
+    run_command,
+)
+
+SlurmSuspendConfig = collections.namedtuple(
+    "SlurmSuspendConfig", ["region", "cluster_name", "boto3_config", "max_batch_size"],
+)
+SUSPEND_LOG_FILE = "/var/log/parallelcluster/slurm_suspend.log"
+
+
+def _get_config():
+    """
+    Get resume program configuration.
+
+    :return: the configuration parameters
+    """
+    logging.info("Reading %s", CONFIG_FILE_PATH)
+
+    config = ConfigParser()
+    config.read(CONFIG_FILE_PATH)
+    lvl = config.get("slurm_cb_config", "loglevel", fallback=logging.INFO)
+    logging.getLogger().setLevel(lvl)
+
+    boto3_config = {}
+    region = config.get("slurm_cb_config", "region")
+    cluster_name = config.get("slurm_cb_config", "cluster_name")
+    max_retry = int(config.get("slurm_cb_config", "max_retry", fallback=DEFAULT_MAX_RETRY))
+    if max_retry > 0:
+        boto3_config["retries"] = {
+            "max_attempts": max_retry,
+            "mode": "standard",
+        }
+    max_batch_size = int(config.get("slurm_cb_config", "max_batch_size", fallback=DEFAULT_MAX_INSTANCES_BATCH_SIZE))
+
+    proxy = config.get("slurm_cb_config", "proxy", fallback="NONE")
+    if proxy != "NONE":
+        boto3_config["proxies"] = {"https": proxy}
+
+    logging.info(
+        (
+            "Configured parameters: region=%s cluster_name=%s boto3_config=%s max_batch_size=%s",
+            region,
+            cluster_name,
+            boto3_config,
+            max_retry,
+            max_batch_size,
+        )
+    )
+    return SlurmSuspendConfig(region, cluster_name, Config(**boto3_config), max_batch_size)
+
+
+def delete_instances(instance_ids_to_nodename, region, boto3_config, batch_size):
+    """Terminate corresponding EC2 instances."""
+    ec2_client = boto3.client("ec2", region_name=region, config=boto3_config)
+    logging.info("Terminating instances %s", list(instance_ids_to_nodename.keys()))
+    for instances in grouper(instance_ids_to_nodename.keys(), batch_size):
+        try:
+            ec2_client.terminate_instances(InstanceIds=instances,)
+        except ClientError as e:
+            logging.error("Failed when terminating instances %s with error %d", instances, e)
+            _set_nodes_to_down([instance_ids_to_nodename[instance_id] for instance_id in instances])
+
+
+def _get_instance_ids_to_nodename(slurm_nodes, region, boto3_config):
+    """Retrieve dict that maps from instance ids to slurm nodenames."""
+    node_ip_to_name = {node.nodeaddr: node.name for node in slurm_nodes}
+    ec2_client = boto3.client("ec2", region_name=region, config=boto3_config)
+    paginator = ec2_client.get_paginator("describe_instances")
+    response_iterator = paginator.paginate(
+        Filters=[{"Name": "private-ip-address", "Values": list(node_ip_to_name.keys())}],
```

Comment:
is the paginator taking care also of handling very long lists?

Also, I would feel safer if we filter by some tag (such as ClusterName), just to make sure we are not terminating for any reason nodes that do not belong to parallelcluster